pkg/apiserver/handlers.go

```diff
@@ -17,21 +20,58 @@ func (s *Server) StartJob(ctx context.Context, req *pb.StartJobRequest) (*pb.Sta
 		// NOTE: Not supposed to happen as already checked, but check anyway.
 		return nil, fmt.Errorf("getUserFromContext: %w", err)
 	}
+	// Contextcheck // False positive. We don't want to use the request context to start the job in the background.
 	jobID, err := s.jobmanager.StartJob(user, req.GetCommand(), req.GetArgs())
 	if err != nil {
-		return nil, fmt.Errorf("jobmanager start job: : %w", err)
+		return nil, fmt.Errorf("job manager start job: : %w", err)
 	}
 	return &pb.StartJobResponse{JobId: jobID.String()}, nil
 }
 
-func (s *Server) StopJob(_ context.Context, _ *pb.StopJobRequest) (*pb.StopJobResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method StopJob not implemented")
+func (s *Server) StopJob(_ context.Context, req *pb.StopJobRequest) (*pb.StopJobResponse, error) {
+	jobID, _ := uuid.Parse(req.GetJobId()) // Already validated in middleware.
+
+	if err := s.jobmanager.StopJob(jobID); err != nil {
+		return nil, fmt.Errorf("job manager stop job: %w", err)
+	}
+
+	return &pb.StopJobResponse{}, nil
 }
 
-func (s *Server) GetJobStatus(_ context.Context, _ *pb.GetJobStatusRequest) (*pb.GetJobStatusResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method GetJobStatus not implemented")
+func (s *Server) GetJobStatus(_ context.Context, req *pb.GetJobStatusRequest) (*pb.GetJobStatusResponse, error) {
+	jobID, _ := uuid.Parse(req.GetJobId()) // Already validated in middleware.
+	job, err := s.jobmanager.LookupJob(jobID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "lookup job: %s", err)
+	}
+	resp := &pb.GetJobStatusResponse{Status: job.Status()}
+	if resp.GetStatus() != pb.JobStatus_JOB_STATUS_RUNNING {
+		exitCode := int32(job.ExitCode()) //nolint:gosec // False positive.
```

Comment:
What is the gosec violation here?

Reply:
ExitCode() returns an int which could overflow int32(). But under the hood, in POSIX, it is uint8, and we don't support windows (which I believe use int32).

Reply:
Thanks, an updated comment does help.

Reply:
Done.